packages/web-components/src/text-input/text-input.styles.ts

```diff
@@ -0,0 +1,252 @@
+  }
+  :host ::slotted([slot='end']) {
+    padding-left: ${spacingHorizontalXXS};
+    gap: ${spacingHorizontalXS};
+  }
+  :host(:hover) .root {
+    border-color: ${colorNeutralStroke1Hover};
+    border-bottom-color: ${colorNeutralStrokeAccessibleHover};
+  }
+  :host(:active) .root {
+    border-color: ${colorNeutralStroke1Pressed};
+  }
+  :host([appearance='outline']:focus-within:not([disabled])) .root {
+    border: ${strokeWidthThin} solid ${colorNeutralStroke1};
+  }
+  :host(:focus-within:not([disabled])) .control {
+    color: ${colorNeutralForeground1};
+  }
+  :host([disabled]) .root {
+    background: ${colorTransparentBackground};
+    border: ${strokeWidthThin} solid ${colorNeutralStrokeDisabled};
+  }
+  :host([disabled]) .control::placeholder,
+  :host([disabled]) ::slotted([slot='start']),
+  :host([disabled]) ::slotted([slot='end']) {
+    color: ${colorNeutralForegroundDisabled};
+  }
+  :host:focus-within) .root {
+    outline: transparent solid 2px;
+  }
+  :host(:focus-within) .root::after {
+    transform: scaleX(1);
+    transition: transform ${durationNormal} ${curveDecelerateMid};
+  }
+  ::selection {
+    color: ${colorNeutralForegroundInverted};
+    background-color: ${colorNeutralBackgroundInverted};
+  }
+  :host([input-size='small']) .control {
+    font-size: ${fontSizeBase200};
+    font-weight: ${fontWeightRegular};
+    line-height: ${lineHeightBase200};
+  }
+  :host([input-size='small']) .root {
+    height: 24px;
+    gap: ${spacingHorizontalXXS};
+    padding: 0 ${spacingHorizontalSNudge};
+  }
+  :host([input-size='small']) ::slotted([slot='start']),
+  :host([input-size='small']) ::slotted([slot='end']) {
+    font-size: ${fontSizeBase400};
+  }
+  :host([input-size='large']) .control {
+    font-size: ${fontSizeBase400};
+    font-weight: ${fontWeightRegular};
+    line-height: ${lineHeightBase400};
+  }
+  :host([input-size='large']) .root {
+    height: 40px;
+    gap: ${spacingHorizontalS};
+    padding: 0 ${spacingHorizontalM};
+  }
+  :host([input-size='large']) ::slotted([slot='start']),
+  :host([input-size='large']) ::slotted([slot='end']) {
+    font-size: ${fontSizeBase600};
+  }
+  :host([appearance='underline']) .root {
+    background: ${colorTransparentBackground};
+    border: 0;
+    border-radius: 0;
+    border-bottom: ${strokeWidthThin} solid ${colorNeutralStrokeAccessible};
+  }
+  :host([appearance='underline']:hover) .root {
+    border-bottom: ${strokeWidthThin} solid ${colorNeutralStrokeAccessibleHover};
+  }
+  :host([appearance='underline']:active) .root {
+    border-bottom: ${strokeWidthThin} solid ${colorNeutralStrokeAccessiblePressed};
+  }
+  :host([appearance='underline']):focus-within .root {
+    border: 0;
+    border-bottom: ${strokeWidthThin} solid ${colorNeutralStrokeAccessiblePressed};
+  }
+  :host([appearance='underline'][disabled]) .root {
+    border-bottom-color: ${strokeWidthThin} solid ${colorNeutralStrokeDisabled};
```

Comment:
you are using ` border-bottom-color` and not `border-bottom`, but you should be using `border-bottom-color` for all these except `:host([appearance='underline']) .root` and just reassign the color.